jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/ls/java/JakartaTextDocuments.java

```diff
@@ -229,11 +230,50 @@ private CompletableFuture<ProjectLabelInfoEntry> getProjectInfo(JakartaTextDocum
                         });
     }
 
+    CompletableFuture<ProjectLabelInfoEntry> getProjectInfoFromClient(JakartaTextDocument document) {
+        String documentURI = document.getUri();
+        // Search future which load project info
+        CompletableFuture<ProjectLabelInfoEntry> projectInfo = null;
+
+        // TODO: re-enable the retrieval of ProjectLabelInfoEntry objects from the caches?
+
+        // We are not going to cache the ProjectLabelInfoEntry for right now - currently caching
+        // is breaking the classpath filtering behaviour - if the pom changes to restrict the classpath, caching
+        // prevents us from returning to the client to re-compute the classpath contents.
+
+        // see getProjectInfoFromCache
+
+        JakartaJavaProjectLabelsParams params = new JakartaJavaProjectLabelsParams();
+        params.setUri(documentURI);
+        params.setTypes(getSnippetRegistry().getTypes());
+        final CompletableFuture<ProjectLabelInfoEntry> future = projectInfoProvider.getJavaProjectLabels(params);
+        ProjectLabelInfoEntry entry = null;
+        try {
+            // future.get() will definitely wait but forced me to re-write the code abit
+            entry = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            // TODO Auto-generated catch block
```

Comment:
The code looks ok, but i would remove all of the comments that are not pertinent here (The ones copied from the method that caches).
Also, since e.printStackTrace() is generally not a good thing to leave in the code, and It will for sure get flagged by app scans, perhaps you can issue a message instead using the LOGGER instance above (i.e. LOGGER.warning("my message: " + e.getMessage()))

Reply:
will do